python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -166,3 +200,125 @@ def _(py_val):
     cdef unique_ptr[scalar] c_obj = make_string_scalar(py_val.encode())
     cdef Scalar slr = _new_scalar(move(c_obj), dtype)
     return slr
+
+
+@singledispatch
+def _from_numpy(np_val):
+    if np_error is not None:
+        raise np_error
+    raise TypeError(f"{type(np_val).__name__} cannot be converted to pylibcudf.Scalar")
+
+
+@_from_numpy.register(np.datetime64)
```

Comment:
Have you tested this code in an environment without numpy? I expect that it will fail at the first register call because np.datetime64 won't exist, nor will any of the subsequent numpy types that you try to access. You'll need to put all of these overloads inside an `if np is not None` block.

Reply:
Good catch, I hadn't tested this without numpy.

Do you think it's worth testing pylibcudf without numpy in CI to ensure this holds?

Reply:
Hmm that's a good question. Let's open an issue around optional dependency testing in pylibcudf. We can address this after we make pyarrow optional since at that point we will have multiple optional dependencies that are also common packages in the ecosystem. I would suggest that we implement some strategy using pytest marks to allow us to run subsets of the test suite in multiple runs i.e. with and without optional deps.